src/resources/checkbox.lua

```diff
@@ -0,0 +1,171 @@
+ 
+checkbox = {
+  parent = Geyser.Container,
+  name = 'CheckboxClass',
+  checkboxLocation = "https://demonnic.github.io/image-assets/checkbox-25px.png",
+  uncheckedLocation = "https://demonnic.github.io/image-assets/unchecked-25px.png"  
+}
+
+checkbox.__index = checkbox
+setmetatable(checkbox, checkbox.parent)
+
+
+--local gss = Geyser.StyleSheet
+local directory = getMudletHomeDir() .. "/checkbox/"
+local saveFile = directory .. "fileLocations.lua"
+if not io.exists(directory) then
+  lfs.mkdir(directory)
+end
+
+
+function checkbox:new(cons, container)
+  cons = cons or {}
+  local consType = type(cons)
+  if consType ~= "table" then
+    printError(f"checkbox:new(cons, container): cons as table of options expected, got {consType}!", true, true)
+  end
+  cons.name = cons.name or Geyser.nameGen("checkbox")
+  local me = self.parent:new(cons, container)
+  setmetatable(me, self)
+  me:createComponents()
+  return me
+end
+
+
+function checkbox:createDisplay()
+
+  self:set_constraints({x = "30%", y = "60%", width = 200, height = 25})
```

Comment:
these should be set via the cons values in the checkbox:new function. We just then let inheritance handle the rest.